xbmc/platform/win32/network/NetworkWin32.cpp

```diff
@@ -47,6 +47,10 @@ bool CNetworkInterfaceWin32::IsConnected() const
 std::string CNetworkInterfaceWin32::GetMacAddress() const
 {
   std::string result;
+
+  if (m_adapter.PhysicalAddressLength < 6)
```

Comment:
magic number? :stuck_out_tongue_winking_eye:

Reply:
The 6 bytes of a mac adress?

Reply:
Yes, because in the next line are read 6 bytes:
```
result = StringUtils::Format("{:02X}:{:02X}:{:02X}:{:02X}:{:02X}:{:02X}", mAddr[0], mAddr[1],
                               mAddr[2], mAddr[3], mAddr[4], mAddr[5]);
```
i.e. `m_adapter.PhysicalAddressLength` is 5 or 0 cannot be read 6 bytes...

Reply:
I guess my point is that this should probably be defined in a human readable way instead of just having `6` littered around 😊

Reply:
Also:

> PhysicalAddressLength
> 
> Type: DWORD
> 
> The length, in bytes, of the address specified in the PhysicalAddress member. For interfaces that do not have a data-link layer, this value is **zero**.

https://docs.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_adapter_addresses_lh

Reply:
Yeah, there is the Microsoft API, which is limited to: MAX_ADAPTER_ADDRESS_LENGTH

But we already use the 6 hardcoded 
here:https://github.com/xbmc/xbmc/blob/4a9f49be4bc2ea8a0057deb48b3cfc6043f04eb0/xbmc/platform/win32/network/NetworkWin32.cpp#L60 - as we exactly expect a 6 byte MAC address.

So - if we want to make the "API check" API compliant, we should use: MAX_ADAPTER_ADDRESS_LENGTH - this is also no problem as kodi internally only uses the first 6 bytes

Reply:
Is not a issue of  NetworkWin32 implementation, is already hardcoded here:

https://github.com/xbmc/xbmc/blob/aa5ec0ceb68ed321c91329838c54204ca014f379/xbmc/network/Network.h#L27

(Network interface for all platforms)

This PR "only" fixes a crash,  it is not intended to change/improve the Kodi network implementation. 
Improvements can be done in another PR.

Reply:
Yeah. That is clear top me. The 6 is hardcoded cause other stuff that this PR does not touch at all need it that way